### ISSUE TITLE
link for the author_id="0" Warning in PublishPress error logs

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Authors#431]
`[PublishPress Authors] Warning - The link for the author_id="0" was changed to the post page because the author is not specified in the given url: https://publishpress.com/author/. permastruct=/author/%author% and home_url=https://publishpress.com`

<img width="601" alt="Screen Shot 2021-04-23 at 9 36 52 AM" src="https://user-images.githubusercontent.com/3868028/115879283-7a6b7d80-a417-11eb-946e-bb1f8f103a5d.png">

**Comment by andergmartins** [comment]
In order to find out what is triggering that behavior that forces to reset the author url, I'm moving that error message inside a  "Is Debug Activated" conditional and adding storing the backtrace.

**Comment by stevejburge** [comment]
@andergmartins This is filling up the TaxoPress.com error logs

It started immediately after I changed the username (and so the profile URL) for an author.

**Comment by andergmartins** [comment]
@stevejburge I wasn't able to replicate it by changing the username. I tried the plugin "Username Changer".

I was only able to replicate it after I forced the `post_author` to `0`. It sounds like it happens when we have corrupted data on that field, or leftovers from the time we allowed to set a post with no author. In those cases, if using our author box, or shortcode, the author names are displayed correctly because we have a fallback in the code that looks for authors based on the relationship between the post ID and our author terms, not necessarily relying on the `post_author` field.

But I've added a conditional and that warning will only be added to the log if `WP_DEBUG` is enabled.

That warning appeared before when the author URL doesn't have the author name, like `https://........./author/`, probably because the `post_author` was empty.
